public/pages/DetectorResults/containers/AnomalyHistory.tsx

```diff
@@ -99,58 +118,70 @@ export const AnomalyHistory = (props: AnomalyHistoryProps) => {
 
   const [selectedHeatmapCell, setSelectedHeatmapCell] = useState<HeatmapCell>();
 
+  const [entityAnomalySummaries, setEntityAnomalySummaries] = useState<
+    EntityAnomalySummaries[]
+  >();
+
+  const [heatmapDisplayOption, setHeatmapDisplayOption] = useState<
+    HeatmapDisplayOption
+  >(INITIAL_HEATMAP_DISPLAY_OPTION);
+
   const detectorCategoryField = get(props.detector, 'categoryField', []);
   const isHCDetector = !isEmpty(detectorCategoryField);
   const backgroundColor = darkModeEnabled() ? '#29017' : '#F7F7F7';
 
-  useEffect(() => {
-    // We load at most 10k AD result data points for one call. If user choose
-    // a big time range which may have more than 10k AD results, will use bucket
-    // aggregation to load data points in whole time range with larger interval.
-    async function getBucketizedAnomalyResults() {
-      try {
-        setIsLoadingAnomalyResults(true);
-        const anomalySummaryResult = await dispatch(
-          searchResults(
-            getAnomalySummaryQuery(
-              dateRange.startDate,
-              dateRange.endDate,
-              props.detector.id
-            )
+  // We load at most 10k AD result data points for one call. If user choose
+  // a big time range which may have more than 10k AD results, will use bucket
+  // aggregation to load data points in whole time range with larger interval.
+  // If entity is specified, we only query AD result data points for this entity.
+  async function getBucketizedAnomalyResults(
+    entity: Entity | undefined = undefined
+  ) {
+    try {
+      setIsLoadingAnomalyResults(true);
+      const anomalySummaryResult = await dispatch(
+        searchResults(
+          getAnomalySummaryQuery(
+            dateRange.startDate,
+            dateRange.endDate,
+            props.detector.id,
+            entity
           )
-        );
+        )
+      );
+
+      setPureAnomalies(parsePureAnomalies(anomalySummaryResult));
+      setBucketizedAnomalySummary(parseAnomalySummary(anomalySummaryResult));
 
-        setPureAnomalies(parsePureAnomalies(anomalySummaryResult));
-        setBucketizedAnomalySummary(parseAnomalySummary(anomalySummaryResult));
-
-        const result = await dispatch(
-          searchResults(
-            getBucketizedAnomalyResultsQuery(
-              dateRange.startDate,
-              dateRange.endDate,
-              1,
-              props.detector.id
-            )
+      const result = await dispatch(
+        searchResults(
+          getBucketizedAnomalyResultsQuery(
+            dateRange.startDate,
+            dateRange.endDate,
+            1,
```

Comment:
I think the `interval` parameter name in `getBucketizedAnomalyResultsQuery()` is kind of confusing. This is not related to detector interval, and more about bucket interval base, which is 1 minute. In [getBucketizedAnomalyResultsQuery()](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/da5ead30a8266f43ff5212d59f18ec75f4559a8b/public/pages/utils/anomalyResultUtils.ts#L362), we split the original time range(endTime - startTime) into MAX_DATA_POINTS(1000) buckets, and since time range is at millisecond level, then we divide by 1 minute * MIN_IN_MILLI_SECS, so that the result value is based on minute level, and then it can be used for `fixed_interval` sent to ES in [code](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/da5ead30a8266f43ff5212d59f18ec75f4559a8b/public/pages/utils/anomalyResultUtils.ts#L389).

After 2nd thought, I think the `interval` parameter can be removed.

Reply:
Does that mean if users select last 1 day's data and the detector interval is 1 hour, I still try to get 1000 buckets instead of 24 buckets?

Reply:
no. This `interval` is not related to detector interval. For example: given a large date range selected, such as 2 weeks, which is 2 * 7 * 24 * 3600 * 1000 milliseconds(endTime - startTime), and we want to only show MAX_DATA_POINTS(1000) data points on chart, then we split above date range in milliseconds into 1000 buckets, then each bucket has `fixed_interval` as (2 * 7 * 24 * 3600 * 1000)/ 1000 = 1209600ms. Since we use [minute as unit](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/da5ead30a8266f43ff5212d59f18ec75f4559a8b/public/pages/utils/anomalyResultUtils.ts#L389) in query to ES, we divide it by 60(sec) * 1000(ms), aka 1209600/(60 * 1000) and then the result is the `fixed_interval` value. This is what `getBucketizedAnomalyResultsQuery()` does.

Reply:
I still don't get it.  Why doesn't getBucketizedAnomalyResultsQuery consider both data range and detector interval?  If there less than MAX_DATA_POINTS anomaly results to show, why do we still query for MAX_DATA_POINTS intervals?

Reply:
Discussed with Yizhe offline.  So we only call this method when there are more than MAX_DATA_POINTS anomaly results to show.